clients/iceberg-views/src/main/java/org/apache/iceberg/nessie/NessieViewCatalog.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (C) 2022 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.iceberg.nessie;
+
+import static org.apache.iceberg.view.ViewUtils.validateTableIdentifier;
+
+import java.util.Map;
+import java.util.function.Function;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.CatalogUtil;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.apache.iceberg.common.DynMethods;
+import org.apache.iceberg.hadoop.HadoopFileIO;
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.view.BaseMetastoreViewOperations;
+import org.apache.iceberg.view.BaseMetastoreViews;
+import org.projectnessie.client.NessieClientBuilder;
+import org.projectnessie.client.NessieConfigConstants;
+import org.projectnessie.client.api.NessieApiV1;
+import org.projectnessie.client.http.HttpClientBuilder;
+import org.projectnessie.error.NessieNotFoundException;
+import org.projectnessie.model.Branch;
+import org.projectnessie.model.ContentKey;
+import org.projectnessie.model.Reference;
+import org.projectnessie.model.TableReference;
+import org.projectnessie.model.Tag;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class NessieViewCatalog extends BaseMetastoreViews implements AutoCloseable {
```

Comment:
I see the need for this at the moment, but it doesn't feel right to keep this long term.
Maybe explicitly mark this as "USE AT YOUR OWN RISK / EXPERIMENTAL CODE" or so.